### PR TITLE
Expand data in init_texture() on arm like update_texture()

### PR DESCRIPTION
**Pull request by sotaroikeda**
Fixes issue #661.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/779)

### CODE REVIEW

File: webrender/src/device.rs [hunk 1]
```diff
@@ -1111,13 +1111,33 @@ impl Device {
...
                 self.set_texture_parameters(texture_id.target, filter);
-                self.upload_texture_image(texture_id.target,
-                                          width,
-                                          height,
-                                          internal_format as u32,
-                                          gl_format,
-                                          type_,
-                                          pixels);
+                let mut expanded_data = Vec::new();
+                if pixels.is_some() &&
```
Comment (kvark) [inline webrender/src/device.rs]: we can rewrite this to be nicer, like
```rust
let actual_pixels = if <same condition> {
    // need to `use std::iter::repeat` at the top
    pixels.unwrap().iter().flat_map(|&byte| repeat(byte).take(4)).collect();
} else {
    pixels
};
self.upload_texture_image(..., actual_pixels);
```

**Comment by sotaroikeda** [comment]
@kvark, I updated a patch. Can you review again?

**Comment by kvark** [comment]
Thanks!
@bors-servo r+

**Pull request merged**